File: relisoft-hr/server/models/OrganizationRole.js
import mongoose from 'mongoose';

const organizationRoleSchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true, trim: true },
  label: { type: String, required: true, trim: true },
  baseRole: {
    type: String,
    enum: ['superadmin', 'admin', 'hr', 'manager', 'employee', 'finance', 'it'],
    default: 'employee',
  },
  roleValue: { type: Number, required: true, unique: true },
  isActive: { type: Boolean, default: true },
}, { timestamps: true });

export default mongoose.model('OrganizationRole', organizationRoleSchema);

File: relisoft-hr/server/Models/OrganizationRole.cs
using System.ComponentModel.DataAnnotations;

namespace RelisoftHR.Models;

public class OrganizationRole
{
    [Key]
    public int Id { get; set; }
    [Required, MaxLength(100)]
    public string Name { get; set; } = "";
    [MaxLength(100)]
    public string? Label { get; set; }
    public bool IsCustom { get; set; }
    public int BaseRoleId { get; set; }
    public int Importance { get; set; }
    public ICollection<Employee> Employees { get; set; } = new List<Employee>();
}
